### ISSUE TITLE
mksurfdata_esmf should use site/grid name rather than "NX x NY" name

### ISSUE THREAD

**Issue by ekluzek** [ESCOMP/CTSM#1791]
In order to be more compatible with single point site names and also all of the different types of global grid names such as the 360x720cru name and the NLDAS2 name, we need mksurfdata_esmf to output names using the users input site name, rather than a name with "NX"x"NY".

**Comment by slevisconsulting** [comment]
Looking again at `gen_mksurfdata_namelist.py` I find that the code does what this issue requests. For example:
- `./gen_mksurfdata_namelist.py --res 5x5_amazon --start-year 1850 --end-year 1850` returns
`surfdata_5x5_amazon_hist_78pfts_CMIP6_1850_c220706.namelist`
- `./gen_mksurfdata_namelist.py --res 360x720cru --start-year 1850 --end-year 1850` returns
`surfdata_360x720cru_hist_78pfts_CMIP6_1850_c220706.namelist`
It uses NX x NY for custom grids such as:
- `./gen_mksurfdata_namelist.py --start-year 2000 --end-year 2000 --model-mesh /glade/work/slevis/data_UCB/ca_fates_input_data/wrfinput_to_meshes/wrfinput_d01_unstruct.nc` returns
`surfdata_109x104_hist_78pfts_CMIP6_1850_c220705.namelist`

Keep in mind that the code obtains mesh_file paths from .../ccs_config/component_grids_nuopc.xml which does not include paths to the 1x1 mesh_files. In a meeting with @ekluzek and @mvertens towards the beginning of the year we agreed that 
- I would test mksurfdata_esmf for 1x1 to confirm that it works. I did that in late March. I had to generate an UNSTRUCT 1x1_brazil destination mesh and add the path to ccs_config/component_grids_nuopc.xml for this to work.
- Also that for the most part we expected to generate fsurdat files for 1x1 using the subset_data tool.

If there's nothing additional here, I think this issue may be closed.

**Comment by ekluzek** [comment]
@slevisconsulting it's those custom cases that need to change. For the WRF case I'd make "res" still required and "res" would be the string that would be used in the surface dataset filename. When people make custom cases they need to be able to have appropriate names for them.

So it would be something like:

```
./gen_mksurfdata_namelist.py --res wrfd01 --start-year 2000 --end-year 2000 --model-mesh /glade/work/slevis/data_UCB/ca_fates_input_data/wrfinput_to_meshes/wrfinput_d01_unstruct.nc returns
surfdata_wrfd01_hist_78pfts_CMIP6_1850_c220705.namelist
```

**Comment by slevisconsulting** [comment]
Closed with merge of #1796
